docs/content/guides/dagster/local_to_production_tutorial.mdx

```diff
@@ -0,0 +1,395 @@
+---
+title: Development to Production Workflow | Dagster
+description: This guide walks through how to transition your code from local development to production
+---
+
+# Development to Production Workflow
+
+In this example, we will walk through how to transition your assets and jobs from local development, to unit testing, and production deployments. We will cover swapping out resources and IO managers and setting run config based on the environment where your code will be executed.
+
+If you’re new to dagster, here’s an overview of the main concepts we’ll be using in this example (if you’re already familiar with dagster, feel free to skip to the next section):
+
+- asset - an asset is a software object that models a data asset. The prototypical example is a table in a database or a file in cloud storage.
+- resource - a resource is an object that models a connection to a (typically) external service. Resources can be shared between assets and different implementations of resources can be swapped in depending on environment. For example, a resource may provide methods to send messages in Slack.
+- IO manager - an IO manager is a special kind of resource that handles storing and loading assets. For example, if we wanted to store assets in S3, we would use dagster’s built in S3 IO manager.
+- run config - assets and resources sometimes require configuration to set certain values (like the password to a database). Run config allows you to set these values at run time. In this example, we will also use an API to set some default run configuration.
+
+## Problem Statement
+
+Let’s say we’ve been tasked with fetching data from Hacker News and splitting it into two datasets: one containing all of the data about stories and one containing all of the data about comments. We have the following additional requirements:
+
+- For local development, we want to store the datasets as tables in a development snowflake database called `SANDBOX` that has a schema for each user.
+- For production, we want to store the datasets as tables in a production snowflake database called `PRODUCTION` that has a schema called `HACKER_NEWS`.
+- For testing, we don’t want to fetch data from Hacker News, instead we want to provide seed data so we can assert that the data was split into the stories and comments datasets correctly.
+
+### Setup
+
+If you want to follow along with this example, you can either:
+
+Clone the example (this will give you the final version of the code):
+
+    dagster new-project --example local_to_prod
+    cd local_to_prod
+    pip install .
+
+Start with a scaffolded project and fill in the code as we go
+
+    dagster new-project --quick-start local_to_prod
+    cd local_to_prod
+    pip install dagster dagit pandas pytest dagster-snowflake dagster-snowflake-pandas
+
+### Part One: Local Development
+
+Let’s start by writing the assets
```

Comment:
This is the summary from the **Problem statement** that describes this section:

> For local development, we want to store the datasets as tables in a development snowflake database called `SANDBOX` that has a schema for each user.

I think it'd be helpful to replace the current sentence (`Let's start by writing the assets`) with something like this that's more specific. It can help add context to the example code that follows.

Reply:
I liked your suggestion of having a list of what we'll do in each section. I added that to the start of this section to hopefully add the missing context